### PR TITLE
Refactors Regal Rats into Basic Mobs (more titles edition)

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

I literally can't focus on anything nowadays, so I just did this to break a never-ending chain of distress. Anyways, regal rats! These fellas are mostly player controlled, but did have _some_ AI capabilities (mainly tied to their actions), so that was incorporated too. Everything should work as-expected (as well as look a shitload cleaner).

Instead of doing weird and awful conditional signals being sent out, I made the `COMSIG_REGAL_RAT_INTERACT` (not the actual name) have a return value so we can always rely on that working whenever we have that signal registered on something we attack. I also cleaned up pretty much every proc related to regal rats, gave them AIs to reflect their kingly nature (and action capabilities (as well as move the action to `mob_cooldown`)).

Since I thought they needed it, Regal Rats now get a special moniker! This is stuff like "the Big Cheese" and what-not, like actual regents in history. That's nice.
## Why It's Good For The Game

Two more off the list. Much better code to read. Way smarter rats with spawning their army as part of a retaliatory assault (war). More sovl with better regal rat names. The list goes on.
## Changelog
:cl:
refactor: Regal Rats have been refactored into basic mobs. They should be a bit smarter and retain their docility (until attacked, in which case you should prepare to get rekt by summoned rats), and properly flee when they can instead of just sit there as you beat them to death. The framework for them interacting with stuff (i.e. opening doors while slobbering on food) is a bit more unified too, now. They also have cooler names too!
/:cl:

FYI: Beyond a few code touchups, I haven't touched the actions at all. I do not believe myself to be enthusiastic about fixing anything involving the actions code as of this moment so that this PR is more overbloated unless it's unbelievably stupid or easy to fix.

### CODE REVIEW

File: code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat_actions.dm [hunk 1]
```diff
@@ -218,7 +36,7 @@
  * This action checks some nearby maintenance animals and makes them your minions.
  * If none are nearby, creates a new mouse.
  */
-/datum/action/cooldown/riot
+/datum/action/cooldown/mob_cooldown/riot
```
Comment (Jacquerel) [inline code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat_actions.dm:39]: Make sure these didn't gain a shared cooldown they didn't have before as a result of this repath
Might need to add `shared_cooldown = null`

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Mostly looks good.

File: code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+	obj_damage = 10
+	melee_damage_lower = 13
+	melee_damage_upper = 15
+	attack_verb_continuous = "slashes"
+	attack_verb_simple = "slash"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+
+	// Slightly brown red, for the eyes
+	// Might be a bit too dim
```
[the remark this answers is not in the document]
Comment (san7890) [reply code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm:32]: ask lemon in the dark

File: code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	SIGNAL_HANDLER
+
+	if(DOING_INTERACTION(src, REGALRAT_INTERACTION) || !allowed_to_attack(target))
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(SEND_SIGNAL(target, COMSIG_RAT_INTERACT, src) & COMPONENT_RAT_INTERACTED)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(isnull(mind))
+		return
```
[the remark this answers is not in the document]
Comment (san7890) [reply code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm:175]: i don't know how they'd get around to doing that seeing as they don't randomly click on shit

[the remark this answers is not in the document]
Comment (Jacquerel) [reply code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm:175]: yeah you would need to add an ai routine for it, dont worry about it
i'll do it myself when I give the syndicate basic mobs a "emag doors you attack" component and the same behaviour

File: code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat_actions.dm (hunk 1 above)
Comment (san7890) [reply code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat_actions.dm:40]: lol why were we treating this like a null value

File: code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jacquerel) [reply code/modules/mob/living/basic/space_fauna/regal_rat/regal_rat.dm:32]: my reasoning that its fine is that nobody is complaining that they cant see anything 🙂

**Pull request merged**